bundles/org.openhab.ui/web/src/components/widgets/system/oh-input.vue

```diff
@@ -21,37 +17,65 @@
 </style>
 
 <script>
+import dayjs from 'dayjs'
+
 import mixin from '../widget-mixin'
 import { OhInputDefinition } from '@/assets/definitions/widgets/system'
 
 export default {
   mixins: [mixin],
   widget: OhInputDefinition,
-  mounted () {
-    delete this.config.value
+  data () {
+    return {
+      pendingUpdate: null
+    }
   },
   computed: {
     value () {
       if (this.config.variable && this.context.vars[this.config.variable] !== undefined) {
         return this.context.vars[this.config.variable]
-      } else if (this.config.showOK && this.context.vars[this.config.value] !== undefined) {
-        return this.context.vars[this.config.value]
+      } else if (this.config.sendButton && this.pendingUpdate !== null) {
+        return this.pendingUpdate
       }
       return (this.config.item) ? this.context.store[this.config.item].state : this.config.defaultValue
+    },
+    valueForDatepicker () {
+      const value = this.value
+      switch (this.config.type) {
+        case 'datepicker':
+          if (Array.isArray(value)) return value
+          const date = new Date(value)
+          if (isNaN(date)) return null
+          return [date]
+        case 'date':
+          return (Date.parse(value) > 0) ? value.split('T')[0] : null
+        default:
+          return null
+      }
     }
   },
   methods: {
     updated (value) {
-      if (this.config.showOK) {
-        this.$set(this.context.vars, this.config.value, value)
+      if (this.config.type === 'texteditor') {
+        value = this.$$(this.$refs.input.$el).find('.text-editor-content')[0].innerHTML
+        if (value === this.value) return
+      }
+      if (this.config.type === 'datepicker' && Array.isArray(value) && this.valueForDatepicker[0].getTime() === value[0].getTime()) return
+      if (this.config.sendButton) {
+        this.$set(this, 'pendingUpdate', value)
       }
       if (this.config.variable) {
         this.$set(this.context.vars, this.config.variable, value)
       }
     },
-    clicked () {
-      if (this.config.item) {
-        this.$store.dispatch('sendCommand', { itemName: this.config.item, cmd: this.context.vars[this.config.value] })
+    sendButtonClicked () {
+      if (this.config.item && this.pendingUpdate) {
+        let cmd = this.pendingUpdate
+        if (this.config.type === 'datepicker' && Array.isArray(cmd)) {
+          cmd = dayjs(cmd[0]).format()
+        }
+        this.$store.dispatch('sendCommand', { itemName: this.config.item, cmd })
```

Comment:
```suggestion
        this.$store.dispatch('sendCommand', { itemName: this.config.item, cmd: cmd })
```

Reply:
That's the shorthand notation, it's fine ;)